client/client_pool/include/client/client_pool/concord_client_pool.hpp

```diff
@@ -124,19 +125,19 @@ class ConcordClientPool {
                            uint64_t seq_num,
                            std::string correlation_id = {},
                            std::string span_context = std::string(),
-                           const bftEngine::RequestCallBack& callback = {});
+                           const concord::client::concordclient::SendCallback& callback = {});
 
   // This method is responsible to get write requests with the new client
   // paramters and parse it to the old SimpleClient interface.
   SubmitResult SendRequest(const bft::client::WriteConfig& config,
                            bft::client::Msg&& request,
-                           const bftEngine::RequestCallBack& callback = {});
+                           const concord::client::concordclient::SendCallback& callback = {});
```

Comment:
Why do we pass a callback in every SendRequest call and not set it once?

Reply:
Good point. This PR deals with replacing the callback type. Refactoring the `SendRequest` can be a follow-up.